app/src/main/java/com/example/tasky/feature_login/presentation/LoginScreen.kt

```diff
@@ -54,16 +87,81 @@ fun LoginScreenContent() {
                     .padding(16.dp)
                     .fillMaxWidth()
             ) {
-                TextBox(hintText = stringResource(R.string.email))
-                TextBox(hintText = stringResource(R.string.password))
-
+                TextBox(
+                    hintText = stringResource(R.string.email),
+                    text = email,
+                    onValueChange = { loginViewModel.onEmailChange(it) },
+                    validator = { it.isValidEmail() }
+                )
+                TextBox(
+                    hintText = stringResource(R.string.password),
+                    text = password,
+                    onValueChange = { loginViewModel.onPasswordChange(it) },
+                    validator = { it.isValidPassword() },
+                    isPasswordField = true
+                )
                 Spacer(modifier = Modifier.height(16.dp))
                 SimpleButton(
                     modifier = Modifier.fillMaxWidth(),
-                    onClick = { /*TODO*/ },
+                    enabled = isFormValid,
+                    onClick = {
+                        // clear focus hides the keyboard
+                        focusManager.clearFocus()
+
+                        authenticationViewModel.logInClicked(
+                            LoginUserInfo(
+                                email = email,
+                                password = password
+                            )
+                        )
```

Comment:
The ViewModel already has all the information about the current email and password, so you could save this parameter here and construct this object in the VM itself. Makes the UI more dumb again 😄